Issue#82 루트 랭킹 조회 - 월별 / 이번달(실시간) / 역대 TOP 100 .by 즐겨찾기 수

92jy38 opened the pull request:
## 개요

1. 파트
즐겨찾기 수에 따른 필터링별 루트 랭킹 TOP 100 조회
- 내림차순 
- 반환은 작성자 닉네임, 제목, 설명, 업데이트날짜( null 이면 등록일)
- 미래시점 예외처리
- 올바르지 않은 년,월 예외처리
- 기존 즐겨찾기 리스트 조회 메서드명 수정 get->search(컨벤션)

## PR 유형

어떤 변경 사항이 있나요?

- [x]  새로운 기능 추가
- [ ]  버그 수정
- [ ]  CSS 등 사용자 UI 디자인 변경
- [x]  코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x]  코드 리팩토링
- [x]  주석 추가 및 수정
- [x]  문서 수정
- [x]  테스트 추가, 테스트 리팩토링
- [ ]  빌드 부분 혹은 패키지 매니저 수정
- [x]  파일 혹은 폴더명 수정
- [ ]  파일 혹은 폴더 삭제

## PR Checklist

PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x]  커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x]  변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

92jy38 committed:
[#82]feat: 즐겨찾기 수 랭킹 조회 월별 / 이번달 실시간 / 역대
- 페이징처리
- 미래시점 불가
- 기존 즐겨찾기 리스트조회 코드 수정

92jy38 committed:
[#82]test: 테스트코드 추가
 - 루트 랭킹 조회 월별 / 이번달 실시간 / 역대

92jy38 committed:
[#82]chore: 에러코드 추가

seongjun1130 reviewed (approved):
전체적으로 JPQL을 통해 쿼리문작성을 조건에 맞게 잘하신것같습니다.
하지만 JPQL은 변경에 너무 취약하고 String으로 조건이 만들어지기에 추후에 DSL 등을 이용해서 개발해보시는것도 추천드립니다. 특히 동적쿼리의 경우 DSL 이 유리한편이 많기때문에 꼭 사용해보시길바랍니다.

92jy38 committed:
[#82]chore: 랭킹조회 코드 수정

92jy38 committed:
[#82]refactor: 루트랭킹 관련 코드 Bookmark -> Route 로 리팩토링

# Conflicts:
#	src/main/java/com/sparta/travelconquestbe/api/route/service/RouteService.java

92jy38 committed:
[#82]chore: 루트 랭킹 조회 쿼리 수정

92jy38 committed:
[#82]chore: 에러코드 수정
